packages/node-core/src/indexer/entities/Metadata.entity.ts

```diff
@@ -14,9 +37,39 @@ export type MetadataRepo = typeof Model & {
   new (values?: unknown, options?: BuildOptions): MetadataModel;
 };
 
-export function MetadataFactory(sequelize: Sequelize, schema: string): MetadataRepo {
+async function checkSchemaMetadata(sequelize: Sequelize, schema: string, chainId: string): Promise<boolean> {
+  try {
+    const res = await sequelize.query<Metadata>(`SELECT * FROM "${schema}"._metadata WHERE key = 'genesisHash'`, {
+      type: QueryTypes.SELECT,
+    });
+    return res[0]?.value === chainId;
+  } catch (e) {
+    return false;
+  }
+}
+
+export async function MetadataFactory(
+  sequelize: Sequelize,
+  schema: string,
+  multichain: boolean,
+  chainId: string
+): Promise<MetadataRepo> {
+  let tableName = '_metadata';
+
+  if (multichain) {
+    const oldMetadataName = await checkSchemaMetadata(sequelize, schema, chainId);
+
+    if (oldMetadataName) {
+      throw new Error(
+        'Found metadata entry with matching chain but wrong table name, must match _metadata_[chainId] syntax for multi-chain indexing.'
+      );
+    }
+
+    tableName = `${tableName}_${chainId}`;
```

Comment:
have we count how long the table name is ? According to rule https://til.hashrocket.com/posts/8f87c65a0a-postgresqls-max-identifier-length-is-63-bytes#:~:text=In%20PostgreSQL%2C%20identifiers%20%E2%80%93%20table%20names,the%20allowed%20length%20of%2063., we can not access 63 bytes.

Reply:
<img width="766" alt="image" src="https://user-images.githubusercontent.com/10431657/202569923-82ad9907-fe6f-46e1-b845-fbad82a1e052.png">